### PR TITLE
S3 - Keep track of read files + option to delete files in writer

### PR DESCRIPTION
**Pull request by rchernanko**

### CODE REVIEW

File: pkg/parquet/reader.go [hunk 1]
```diff
@@ -30,6 +30,8 @@ type ReadResult map[string]interface{}
 type ReadResults []ReadResult
 type ResultCallback func(progress Progress, results interface{}) (bool, error)
 
+var S3FilesToDelete []string
```
Comment (rchernanko) [inline pkg/parquet/reader.go]: I don't like this, but essentially need a way of the writer being able to access the files to delete. Alternatively, I could add the file name onto each spend event item, and then keep track of the files to delete that way... Then I could just pass in a slice to the DeleteFilesInS3...

File: pkg/parquet/reader.go [hunk 2]
```diff
@@ -156,6 +158,7 @@ func (r *s3Reader) ReadDateAsync(ctx context.Context, datetime time.Time, target
 				stop = true
 				return
 			}
+			S3FilesToDelete = append(S3FilesToDelete, file)
```
Comment (floelm) [inline pkg/parquet/reader.go]: this should be rather something like `readS3Files` and then depending on the options delete the fiiles on read date right?

Comment (rchernanko) [reply pkg/parquet/reader.go]: That is a very good point. Will update the name

File: pkg/parquet/reader.go (hunk 1 above)
Comment (rchernanko) [reply pkg/parquet/reader.go]: will discuss with Flo in chat

File: pkg/parquet/writer.go [hunk 3]
```diff
@@ -185,3 +193,37 @@ func makeTags(tags map[string]string) []*s3.Tag {
 
 	return s3Tags
 }
+
+func (w *s3Writer) DeleteReadFiles(ctx context.Context) error {
```
Comment (floelm) [inline pkg/parquet/writer.go]: DeleteReadFiles and DeleteWrittenFiles look quite similar, can we reuse something?

**Pull request closed without merging**